includes/class-wc-abstract-google-analytics-js.php

```diff
@@ -57,45 +61,44 @@ public static function header() {
 			function gaOptout() {
 				document.cookie = disableStr + '=true; expires=Thu, 31 Dec 2099 23:59:59 UTC; path=/';
 				window[disableStr] = true;
-			}
-		</script>";
+			}";
+
+		wc_enqueue_js( $code );
```

Comment:
I apply this he name change here https://github.com/woocommerce/woocommerce-google-analytics-integration/pull/260/commits/4b97db4f5d7c89529374da46947109083b6907fb ... however, if I run it via register/add_inline script. It stops working and the script is not being loaded 🤔

Reply:
In this case is not important to have this one in the footer. Since it's a function that should be called by something... 

`<a href="javascript:gaOptout()">Click here to opt-out of Google Analytics</a>`

But seems is not being used... maybe it was in the past? 🤔

Reply:
That's because it also needs a `wp_enqueue_script` to force it to appear. However my suggestion was to add it as a dependency for the `google_tag_manager` script (right now we pass an empty array of dependencies. Making it a dependency forces it to be included before (which is where it used to be).

Ex:
```php
wp_register_script(
	'google-tag-manager',
	'https://www.googletagmanager.com/gtag/js?id=' . esc_js( $gtag_id ),
	array( 'google-analytics-opt-out' ),
	WC_GOOGLE_ANALYTICS_INTEGRATION_VERSION,
	false
);
```

For the `gaOptout()` function it's ok to have it in the footer, but what that does is set a cookie, which is checked on any subsequent pages to disable tracking. If we check the cookie and disable the tracking in the footer then that's too late.

This is the part that must run before the tracking code:
```js
if ( document.cookie.indexOf( disableStr + '=true' ) > -1 ) {
	window[disableStr] = true;
}
```